ee/tabby-webserver/src/service/email/testutils.rs

```diff
@@ -0,0 +1,63 @@
+use std::time::Duration;
+
+use serde::Deserialize;
+use tabby_db::DbConn;
+use tokio::process::{Child, Command};
+
+use super::new_email_service;
+use crate::schema::email::{AuthMethod, EmailService, EmailSettingInput, Encryption};
+
+#[derive(Deserialize, Debug)]
+pub struct Mail {
+    pub sender: String,
+    pub subject: String,
+}
+
+pub struct TestEmailServer {
+    #[allow(unused)]
+    child: Child,
+}
+
+impl TestEmailServer {
+    pub async fn list_mail(&self) -> Vec<Mail> {
+        let mails = reqwest::get("http://localhost:1080/api/messages")
+            .await
+            .unwrap();
+
+        mails.json().await.unwrap()
+    }
+
+    async fn init_service(&self, db_conn: DbConn) -> impl EmailService {
+        let service = new_email_service(db_conn).await.unwrap();
+        service
+            .update_email_setting(default_email_settings())
+            .await
+            .unwrap();
+        service
+    }
+
+    pub async fn start(db_conn: DbConn) -> (TestEmailServer, impl EmailService) {
```

Comment:
```suggestion
    pub async fn new() -> TestEmailServer {
```

Reply:
I want this to be doing both in one step, to ensure that you don't forget to initialize the service by providing the email settings. It takes the database connection so that it's not always creating its own, which would cause problems in the auth service tests.

Reply:
still preferring separating them since the real dependency of db_conn is email service, not test email server it self.